### DISCUSSION TITLE
How to import "relation fields" to use with model types

### DISCUSSION THREAD

**Discussion by taylorhlabs** [prisma/prisma#6811] [Q&A] [answered]
I am using JSDoc with VS Code (but not typescript!) to get some type safety/hinting across my app. I am trying to leverage the types generated by Prisma client and I have it mostly working, but I can't get the "relation fields" to show up in the autocomplete menu or be included in my types. Based on the docs [here](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#type-definitions), the types generated by Prisma do not include relation fields but I'd really like to use them. How can I import the relation fields (or create them?) for my use case? 

This is how I'm using Prisma types with JSDoc currently:
```
import { User } from '@prisma/client/index';
import { Prisma } from '@prisma/client'; // not used yet...
...
/**
* @type {User}
*/
const newUser = { email: 'test@test.com, ... }; // standard properties like 'email' work in autocomplete but not relation fields like 'posts'
```

Note that I've tried using something like `@type {User & Prisma.UserInclude}` which gets me the relation fields in the autocomplete but they're not typed correctly for my use case.

**Comment by taylorhlabs** [comment, marked as answer]
Using the docs [here](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types), I came up with the following to get autocomplete and type checking in my non-typescript project using VS Code:
```
// lib/models/types.ts
import { Prisma } from '@prisma/client';

const userWithRelations = Prisma.validator<Prisma.UserArgs>()(
  {
    include: { posts: true, likes: true }
  }
);

export type UserWithRelations = Prisma.UserGetPayload<
  typeof userWithRelations
>;
```
```
// someFile.js
/**
 * @typedef {import('lib/models/types').UserWithRelations} UserWithRelations
 */
...
 /**
 * @type {UserWithRelations}
 */
const newUser = { ...user }; // autocomplete and type checking works on newUser object
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
